### PR TITLE
Prevent loading entire users table into memory

### PR DESCRIPTION
**Pull request by tylerparsons**
Current implementations of several organization level statistics computations using the Sequel ORM lead to huge performance bottlenecks for organizations with many users.

Many queries, such as `sum` and `count` operations, which can be performed as simple SQL queries, are implemented by rails loading all user records for an organization into memory and perform aggregations in ruby.  This does not scale to large organizations, and causes unnecessary streaming of dataover the network, thus these are rewritten as optimized SQL queries.

This pull request does not cover every organization level query subject to this issue, but it does address the highest hit queries and achieves significant performance gain.

### CODE REVIEW

**Comment by vishalya** [comment]
Looks good.

File: app/models/organization.rb [hunk 1]
```diff
@@ -230,7 +230,12 @@ def get_obs_general_calls(options = {})
   def get_twitter_imports_count(options = {})
     date_from, date_to = quota_dates(options)
 
-    SearchTweet.get_twitter_imports_count(users_dataset.join(:search_tweets, :user_id => :id), date_from, date_to)
+    org_search_tweets = Sequel::Model.db.fetch("
+      SELECT * FROM search_tweets st
```
Comment (vmanohar1) [inline app/models/organization.rb:234]: Do you really need select * here??
We are not having any tweets usage for org users currently but down the road we will, hence asking..

**Comment by vmanohar1** [comment]
Rest looks ok..

File: app/models/organization.rb (hunk 1 above)
Comment (tylerparsons) [reply app/models/organization.rb:234]: This change is mainly to enforce that the inner join with `search_tweets` is always performed, which should return 0 rows.  We should eventually refactor `SearchTweet.get_twitter_imports_count` to manually construct the entire query but I think that change is lower priority given that we currently don't use `search_tweets`.

Comment (vmanohar1) [reply app/models/organization.rb:234]: Agreed.. We can take care of such things when we use twitter..

**Pull request merged**